packages/front-end/components/Features/ConditionInput.tsx

```diff
@@ -40,6 +43,32 @@ export default function ConditionInput(props: Props) {
     setSimpleAllowed(jsonToConds(value, attributes) !== null);
   }, [value, attributes]);
 
+  const savedGroupValues = [
+    {
+      label: "is in the saved group",
+      value: "$inGroup",
+    },
+    {
+      label: "is not in the saved group",
+      value: "$notInGroup",
+    },
+  ];
+
+  const savedGroupOperators: { label: string; value: string }[] = [];
+
+  if (savedGroups) {
+    for (const group of savedGroups) {
+      // map the attributeKey to the attributes map and get the value.datatype associated with that attributeKey.
+      // then, add that datatype as a property to savedGroupOperators where it's associated value is equal to savedGroupValues
+      const attribute = group.attributeKey;
+
+      if (attributes.has(attribute)) {
+        const datatype = attributes.get(attribute).datatype;
+        savedGroupOperators[datatype] = savedGroupValues;
```

Comment:
This doesn't match up with the type declared above.  You are defining `savedGroupOperators` as an array, but then using it like an object.

Also, I don't think this logic is correct.  Say you have 5 different attributes with type `string`, but only one of them has an associated saved group.  We only want the saved group operators to show up for that one, not all 5.

A simpler approach is to use the `savedGroupOptions` you are defining below.

```ts
const savedGroupOperators = [
  {
    label: "is in the saved group",
    value: "$inGroup",
  },
  {
    label: "is not in the saved group",
    value: "$notInGroup",
  },
];

// Later
{ label: "does not exist", value: "$notExists" },
...(savedGroupOptions ? savedGroupOperators : []),

```